p4include/v1model.p4

```diff
@@ -117,8 +213,19 @@ extern action_profile {
     action_profile(bit<32> size);
 }
 
-// Get a random number in the range lo..hi
+/***
+ * Generate a random number in the range lo..hi, inclusive, and write
+ * it to the result parameter.
+ *
+ * TBD: Do p4c and behavioral-model support run-time variable values
```

Comment:
You can read `hi` and `lo` from registers in theory, so they can be set at runtime. They can also come from packet headers or metadata.
The implementation is here: https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution/uniform_int_distribution. At the moment `hi` and `lo` have to fit in 64-bit numbers (but that could be generalized fairly easily). Behavior is undefined if `lo > hi` (see https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution/uniform_int_distribution), but a check could be added.

Reply:
I have another commit for the documentation that explicitly makes the value written to the out parameter unspecified if lo > hi.  I could imagine adding a run-time warning in BMv2, but at least with that behavior documented, people will have some kind of forewarning.